File: 1396-count-servers-that-communicate/count-servers-that-communicate.java
class Solution {
    public int countServers(int[][] g) {
        int n = g.length;
        int m = g[0].length;
        int[] clm = new int[m];
        int[] rw = new int[n] ;
        for(int i = 0 ; i < n ; i ++)
            for(int j = 0 ; j < m ; j ++)
                if(g[i][j] == 1){
                    clm[j] ++ ;
                    rw[i] ++  ;
                }
        
        int ans = 0 ;
        for(int i = 0 ; i < n ; i ++){
            for(int j = 0 ; j < m ; j ++){
                if(g[i][j] == 1 && (clm[j] > 1 || rw[i] > 1)){
                    ans ++ ;
                    // System.out.println(i + " " + j);
                }
            }
        }
        return ans ;
    }
}

File: 1396-count-servers-that-communicate/count-servers-that-communicate.py
class Solution:
    def countServers(self, g: List[List[int]]) -> int:
        n, m = len(g), len(g[0])
        rw, clm = [0] * n, [0] * m
        for i in range(n):
            for j in range(m):
                if g[i][j]:
                    rw[i] += 1
                    clm[j] += 1
        
        ans = 0 
        for i in range(n):
            for j in range(m):
                if g[i][j] and (clm[j] > 1 or rw[i] > 1):
                    ans += 1

        return ans 